### ISSUE TITLE
Deploy to vercel

### ISSUE THREAD

**Issue by ternaus** [lancedb/lancedb#650]
Code works locally on OS X, but throws an error when deployed at Vercel.

`@lancedb/vectordb-linux-x64-gnu` is part of `package.json` and is installed.

```
⨯ Error: vectordb: failed to load native library.
  You may need to run `npm install @lancedb/vectordb-linux-x64-gnu`.

  If that does not work, please file a bug report at https://github.com/lancedb/lancedb/issues
      
  Source error: Error: Cannot find module '@lancedb/vectordb-linux-x64-gnu'
Require stack:
- /var/task/node_modules/vectordb/native.js
- /var/task/node_modules/vectordb/dist/query.js
- /var/task/node_modules/vectordb/dist/remote/index.js
- /var/task/node_modules/vectordb/dist/index.js
- /var/task/.next/server/pages/api/image/similar_faces.js
- /var/task/node_modules/next/dist/compiled/next-server/server.runtime.prod.js
- /var/task/___next_launcher.cjs
    at Object.<anonymous> (/var/task/node_modules/vectordb/native.js:26:11)
    at Module._compile (node:internal/modules/cjs/loader:1256:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
    at Module.load (node:internal/modules/cjs/loader:1119:32)
    at Module._load (node:internal/modules/cjs/loader:960:12)
    at r.<computed>.e._load (/var/task/___vc/__launcher/__launcher.js:14:2299)
    at Module.require (node:internal/modules/cjs/loader:1143:19)
    at w.require (/var/task/node_modules/next/dist/compiled/next-server/server.runtime.prod.js:11:25777)
    at require (node:internal/modules/cjs/helpers:119:18)
    at Object.<anonymous> (/var/task/node_modules/vectordb/dist/query.js:28:25)
Error: vectordb: failed to load native library.
  You may need to run `npm install @lancedb/vectordb-linux-x64-gnu`.

  If that does not work, please file a bug report at https://github.com/lancedb/lancedb/issues
      
  Source error: Error: Cannot find module '@lancedb/vectordb-linux-x64-gnu'
Require stack:
- /var/task/node_modules/vectordb/native.js
- /var/task/node_modules/vectordb/dist/query.js
- /var/task/node_modules/vectordb/dist/remote/index.js
- /var/task/node_modules/vectordb/dist/index.js
- /var/task/.next/server/pages/api/image/similar_faces.js
- /var/task/node_modules/next/dist/compiled/next-server/server.runtime.prod.js
- /var/task/___next_launcher.cjs
    at Object.<anonymous> (/var/task/node_modules/vectordb/native.js:26:11)
    at Module._compile (node:internal/modules/cjs/loader:1256:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
    at Module.load (node:internal/modules/cjs/loader:1119:32)
    at Module._load (node:internal/modules/cjs/loader:960:12)
    at r.<computed>.e._load (/var/task/___vc/__launcher/__launcher.js:14:2299)
    at Module.require (node:internal/modules/cjs/loader:1143:19)
    at w.require (/var/task/node_modules/next/dist/compiled/next-server/server.runtime.prod.js:11:25777)
    at require (node:internal/modules/cjs/helpers:119:18)
    at Object.<anonymous> (/var/task/node_modules/vectordb/dist/query.js:28:25)
Error: Runtime exited without providing a reason
Runtime.ExitError
```

**Comment by Obapelumi** [comment]
Running into this too while trying to deploy to AWS Lambda. Did you by any chance resolve this?

**Comment by poupalov** [comment]
@Obapelumi I could solve this issue for Lambda and made a small demo of this: [vectordb-js-lambda](https://github.com/poupalov/vectordb-js-lambda).

The key to making it work was to
- creating a Lambda layer to provide the right Linux binary to my Lambda function (i.e. `@lancedb/vectordb-linux-x64-gnu`)
- marking `vectordb` as an external dependency when bundling my Lambda function with `esbuild`

cc @ternaus - maybe this approach can also work for Vercel

**Comment by andreia-oca** [comment]
Are there any plans to fix this upstream in the `lancedb` codebase?
My team and users are also affected by this and it's pretty annoying to implement hacks.

What I've previously seen in the `npm/nodejs` ecosystem is that libraries are making use of `node-pre-gyp` to handle binary cross builds. Essentially, using `node-pre-gyp` will give your users the ability to execute:
```
node-pre-gyp install --target_platform=linux --target_arch=arm64 vectordb
```

**Comment by westonpace** [comment]
@andreia-oca can you help me understand this problem?  The `vectordb` package lists `@lancedb/vectordb-linux-x64-gnu` (and all the other platform-specific variations) as optional dependencies in the `package.json` file.  At install time, npm should try and install all optional dependencies.  It will fail to download the variations that don't make sense on the target but should download the correct variation for the target.

In other words, after running `npm install vectordb` you should have both `vectordb` and `@lancedb/vectordb-linux-x64-gnu` if you are running on an x64 system.

Why does this fail to succeed in your environment?  Can you run `npm install --loglevel verbose vectordb` and upload the log to this ticket?

**Comment by andreia-oca** [comment]
So, my use case is the following:

I am building and bundling the npm dependencies for my project locally on darwin-arm64. This basically means that I will install `@lancedb/vectordb-darwin-arm64`.

I am archiving everything and deploying it to a lambda instance that can only be `linux-x86` or `linux-arm64`. Hence, the code I've built locally will not work remotely on Lambda, it will fail with `install @lancedb/vectordb-linux-arm64`.

**Comment by westonpace** [comment]
@andreia-oca 

Thank you, I understand now.  I believe that very recent versions of npm now actually support specifying a target os/arch.  Can you try the following?

```
# You may need to use a more recent version of node 18 for this command to work
npm install npm@latest
npm install --os darwin --cpu arm64 vectordb
```

Alternatively, if you have a `node-pre-gyp` solution that doesn't require making users install all the different binaries (e.g. we don't want to download all binaries and then pick the right one at runtime because each binary is ~50MB) then we would be open to a PR.

**Comment by westonpace** [comment]
Note, if your `npm` version is not recent enough the `--os` and `--cpu` flags won't fail they will simply be ignored (and you will get some error about no package named `arm64`

**Comment by PrettySolution** [comment]
Make sure you've installed most resent version https://www.npmjs.com/package/vectordb
Do not blindly copy `npm install` commands from tutorials :)

**Comment by PrettySolution** [comment]
I nave the same issue building NodejsFunction in aws-cdk. 

How did you guys resolved?

**Comment by malikalimoekhamedov** [comment]
I experience the same issue with version 0.10.0.
